satellite/overlay/selectednodescache.go

```diff
@@ -0,0 +1,183 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package overlay
+
+import (
+	"context"
+	"math/rand"
+	"sync"
+
+	"go.uber.org/zap"
+	"storj.io/common/storj"
+)
+
+// CachedNode contains all the info about a node in the cache
+// The info we need about a node in the cache is the minimum info
+// to create an order limit
+type CachedNode struct {
+	ID             storj.NodeID
+	LastNet        string
+	LastIPPort     string
+	MinimumVersion string // semver or empty
+}
+
+// SelectedNodesCache keeps a list of all the storage nodes qualified to store data
+type SelectedNodesCache struct {
+	log              *zap.Logger
+	db               DB
+	vetted           []CachedNode
+	vettedMu         sync.Mutex
```

Comment:
what do you mean by "infrequent system updates"?

Reply:
also, im reading up on atomic value and sounds like it might be better to stick with mutexes... "These functions require great care to be used correctly" re: https://golang.org/pkg/sync/atomic/

Maybe we can start with mutexes and see if we need better performance. If so then consider using atomic value?

Reply:
Ssure, we can improve that later.

"These functions require great care to be used correctly" of course applies to mutexes, channels, goroutines as well.